### DISCUSSION TITLE
Opposite Z order than expected for iso/hex tiles that extend past grid height

### DISCUSSION THREAD

**Discussion by gak** [StarArawn/bevy_ecs_tilemap#257] [Q&A] [answered]
The rendered tiles:
![image](https://user-images.githubusercontent.com/31338/188306798-961e227b-555f-4a8e-bf6d-f9a5e1f5ec9f.png)

The lower tiles should render above the higher tiles, right? Not sure if this is a bug or my set up. I'm using the current main branch, and the same happens on 0.7.0.

Here's the atlas. I've picked out the 3 different coloured cubes. These are 32x32 tiles with a grid size of 32x16:
![image](https://user-images.githubusercontent.com/31338/188306899-36d6431a-611c-4e11-8901-ab96a5b6ffca.png)

I looked at the shader and there is the Z axis being set from 0..1 depending on the Y axis, which to me seems like the right thing to do to.

https://github.com/StarArawn/bevy_ecs_tilemap/blob/6a6fb43e3843f601a635d63abc2ee4411a3876b3/src/render/shaders/staggered_iso.wgsl#L30

Specifically this argument: `1.0 - (tile_z.y / tilemap_data.map_size.y)`.

I wasn't sure if Y was going the wrong direction along with Z, so I reversed the ordering by trying just `tile_z.y / tilemap_data.map_size.y` with the same result. I also messed around with bigger Z values with interesting results (where Z left view frustum) but nothing how I wanted.

Also tried messing with `Camera.depth_calculation` and `Orthographic.depth_calculation` without any luck.

I copied the basic example with a few changes. Here's the startup system:

```rust
pub fn init(mut commands: Commands, asset_server: Res<AssetServer>) {
    commands.spawn_bundle(Camera2dBundle {
        ..Default::default()
    });

    let texture_handle: Handle<Image> =
        asset_server.load("isometric-sandbox-32x32/isometric-sandbox-sheet.png");

    let tilemap_size = TilemapSize { x: 4, y: 4 };
    let tilemap_entity = commands.spawn().id();
    let mut tile_storage = TileStorage::empty(tilemap_size);

    for x in 0..tilemap_size.x {
        for y in 0..tilemap_size.y {
            let tile_pos = TilePos { x, y };

            let texture = match thread_rng().gen_range(0..3) {
                0 => 0,
                1 => 18,
                2 => 36,
                _ => unreachable!(),
            };

            let tile_entity = commands
                .spawn()
                .insert_bundle(TileBundle {
                    position: tile_pos,
                    tilemap_id: TilemapId(tilemap_entity),
                    texture: TileTexture(texture),
                    visible: Default::default(),
                    color: Default::default(),
                    ..Default::default()
                })
                .id();
            tile_storage.set(&tile_pos, Some(tile_entity));
        }
    }

    let tile_size = TilemapTileSize { x: 32.0, y: 32.0 };
    let grid_size = TilemapGridSize { x: 32.0, y: 16.0 };

    commands
        .entity(tilemap_entity)
        .insert_bundle(TilemapBundle {
            grid_size,
            size: tilemap_size,
            storage: tile_storage,
            texture: TilemapTexture(texture_handle),
            tile_size,
            map_type: TilemapType::Isometric {
                neighbors_include_diagonals: true,
                coord_system: IsoCoordSystem::Staggered,
            },
            ..Default::default()
        });
}
```

**Comment by Affinator** [comment]
And to add another point: The tiles are always centered on the grid. If I look at this tileset/atlas, it may be wanted that the top of the cube aligns with the grid. It has the same shape as a 2D-Tile at least.

**Comment by bzm3r** [comment]
@gak We have made some changes to the renderer now. Can you try again and tell me what happens? (I don't think the issue is fixed, but I want to see what has changed, and I want to try and better understand the troubles you were running into, in the context of the current state of the code.)

**Comment by bzm3r** [comment, marked as answer]
@gak Oh, also, I just realized...the tiles do not have z-position at all. It is ignored. It is not used in any meaningful way. If you want to stack tiles, you have to use separate tilemaps. All tiles in the same tilemap have the same z-value.

**Reply by gak** [reply]
Thanks, I suspected as much after my futile attempts. :)

**Reply by bzm3r** [reply]
@gak yeah, sorry about this. We'll be working on implementing this feature next! https://github.com/StarArawn/bevy_ecs_tilemap/discussions/200#discussioncomment-3216154
